### PR TITLE
boards: lilygo: ttgo_lora32: enable SDHC support

### PR DESCRIPTION
**Pull request by lfelten**
enable support for the SDHC controller to use the micro SD card slot

### CODE REVIEW

**Review by danieldegrasse — COMMENTED** [review]
Can you add the `sdhc` tag to the list of features this board supports so the SDHC tests run on it? https://github.com/zephyrproject-rtos/zephyr/blob/73a3438b8204a61211b5f78786799a8e6ba77afa/boards/nxp/lpcxpresso55s69/lpcxpresso55s69_lpc55s69_cpu0.yaml#L27

Also, we should update the documentation page for this board to indicate SDHC support

**Comment by lfelten** [comment]
Thanks for you feedback!

> Can you add the `sdhc` tag to the list of features this board supports so the SDHC tests run on it?
added

> Also, we should update the documentation page for this board to indicate SDHC support
It mentioned SD before, I updated it to SDHC

**Review by sylvioalves — APPROVED** [review]

**Review by sylvioalves — CHANGES_REQUESTED** [review]
Now, add ' sdhc'  to the ignore list in `ttgo_lora32_esp32_appcpu.yaml`.

File: boards/lilygo/ttgo_lora32/ttgo_lora32-pinctrl.dtsi [hunk 1]
```diff
@@ -21,6 +21,18 @@
 		};
 	};
 
+	spim2_default: spim2_default {
```
Comment (raffarost) [inline boards/lilygo/ttgo_lora32/ttgo_lora32-pinctrl.dtsi:24]: this is not related to SDHC, right?

Comment (lfelten) [reply boards/lilygo/ttgo_lora32/ttgo_lora32-pinctrl.dtsi]: Hello Raffael,
no, thankse.  I removed that.

**Comment by lfelten** [comment]
> Now, add ' sdhc' to the ignore list in `ttgo_lora32_esp32_appcpu.yaml`.

Added it. 
I also added CONFIG_ESP32_USE_UNSUPPORTED_REVISION=y in the procpu defconfig, which is required since I rebased to main

**Review by raffarost — DISMISSED** [review]

**Comment by lfelten** [comment]
This fails during linking, undefined symbols are ctime_r and asctime_r.

**Comment by sylvioalves** [comment]
> This fails during linking, undefined symbols are ctime_r and asctime_r. I'll rebase to main.

Indeed. Fix in #79455.

**Comment by sylvioalves** [comment]
@lfelten please rebase.

**Review by danieldegrasse — DISMISSED** [review]

**Comment by sylvioalves** [comment]
@lfelten Is this completed?

File: boards/lilygo/ttgo_lora32/doc/index.rst [hunk 2]
```diff
@@ -205,7 +205,7 @@ message in the monitor:
    ***** Booting Zephyr OS vx.x.x-xxx-gxxxxxxxxxxxx *****
    Hello World! ttgo_lora32
 
-LoRa samples
+Code samples
```
Comment (sylvioalves) [inline boards/lilygo/ttgo_lora32/doc/index.rst:208]: @kartben Sounds this file was not updated to latest doc design. Should we do it before v4.0.0?

Comment (kartben) [reply boards/lilygo/ttgo_lora32/doc/index.rst:208]: sure, we can try :) #81411

Comment (lfelten) [reply boards/lilygo/ttgo_lora32/doc/index.rst:198]: It didn't build out of the box because the device tree lacked "disk-name = "SD";".
I added it, it now builds, and I tested it with the "fs_sample".
Also rebased on to v4.0.0

File: boards/lilygo/ttgo_lora32/ttgo_lora32_esp32_procpu_defconfig [hunk 3]
```diff
@@ -8,3 +8,4 @@ CONFIG_UART_CONSOLE=y
 
 CONFIG_GPIO=y
 CONFIG_I2C=y
+CONFIG_ESP32_USE_UNSUPPORTED_REVISION=y
```
Comment (kartben) [inline boards/lilygo/ttgo_lora32/ttgo_lora32_esp32_procpu_defconfig:11]: Can you add a comment on why this is needed (if this is needed?). This way it's not only clearer to the user but also helps knowing when we might be able to remove it in the future

File: boards/lilygo/ttgo_lora32/doc/index.rst [hunk 4]
```diff
@@ -220,6 +220,26 @@ To build the LoRa receive sample application using sysbuild use the command:
...
+
+The following sample application displays numbers on the OLED display:
+
+.. zephyr-app-commands::
+   :tool: west
+   :zephyr-app: samples/subsys/display/cfb
+   :board: ttgo_lora32/esp32/procpu
+   :goals: build
+   :west-args: --sysbuild
+   :compact:
```
Comment (kartben) [inline boards/lilygo/ttgo_lora32/doc/index.rst:241]: Not blocking but I'd rather you link to the samples instead. I understand that this is useful to give ready to copy-paste build instructions but it's also content that might be left behind should something change in the way these samples should be built (arguably unlikely, but still)
I would simply say something like "You may also run :zephyr:code-sample:`XXX` to blah blah or ... to ...".

**Review by kartben — CHANGES_REQUESTED** [review]
A nit and an actual request for change. Thanks for rebasing, too!

File: boards/lilygo/ttgo_lora32/doc/index.rst (hunk 4 above)
Comment (lfelten) [reply boards/lilygo/ttgo_lora32/doc/index.rst:241]: Hi @kartben 
Yes, I can change them to be links.
Since I usually compare with documentation files from other boards, is there a reference documentation you'd recommend? Especially how to make the links.
The copy-paste build instructions did actually change not so long ago, and I had to update them. So, removing those would be good.

Comment (lfelten) [reply boards/lilygo/ttgo_lora32/doc/index.rst:241]: Should I add links like this?

* :zephyr:code-sample:`lora-send`
* :zephyr:code-sample:`lora-receive`
* :zephyr:code-sample:`fs-fs-sample`
* :zephyr:code-sample:`display-cfb`

(how do subdirectories/catgories get handled?)

File: boards/lilygo/ttgo_lora32/ttgo_lora32_esp32_procpu_defconfig (hunk 3 above)
Comment (kartben) [reply boards/lilygo/ttgo_lora32/ttgo_lora32_esp32_procpu_defconfig:11]: At a minimum, makes this a different (and first) commit, similar to https://github.com/zephyrproject-rtos/zephyr/commit/956edd1c3dbf539f1e4b472c61e78668c5e85a16 since my understanding is that it is an unrelated (but needed) change

Comment (kartben) [reply boards/lilygo/ttgo_lora32/ttgo_lora32_esp32_procpu_defconfig:11]: fwiw i was not suggesting to add a comment to the git commit (although it's not a bad idea to do it too :) ) but rather right in the defconfig file so that people understand where this is coming from. BTW I am not sure this really belongs to the defconfig as this is not something that should be modifiable, no? Shouldn't it be `select`ed in the Kconfig file instead? cc @sylvioalves

Comment (lfelten) [reply boards/lilygo/ttgo_lora32/ttgo_lora32_esp32_procpu_defconfig:13]: It says "unsupported revision" - at least my board uses this version of the chip.
Maybe newer boards (it's still in production) use a newer version.
I'm not sure this setting has any impact when selected for a chip revision > 1

 I added a comment and also trimmed the commit log line length.

Comment (sylvioalves) [reply boards/lilygo/ttgo_lora32/ttgo_lora32_esp32_procpu_defconfig:13]:
```
config ESP32_USE_UNSUPPORTED_REVISION
	bool "Use unsupported ESP32 revision (Rev 0/1)"
	help
	  ESP32 SoC has multiple revisions, some of which are not supported by the current
	  implementation, as such as REV0 and REV1. In case those revisions are required,
	  set this option to enable support for them. Note that this is not recommended and
	  may lead to unexpected behavior.
```

rev0 and rev1 ESP32 SocS (old versions) have rom bugs that we haven't and will not address. So if the board used such old chip revision, this config must be set.

**Review by sylvioalves — APPROVED** [review]

**Review by kartben — APPROVED** [review]
